### PR TITLE
WIP Hackathon Programmatic Certs

### PR DESCRIPTION
**Pull request by caesar2164**
- draw border programmatically

### CODE REVIEW

File: gen_cert.py [hunk 1]
```diff
@@ -1853,3 +1855,453 @@ def _generate_v3_dynamic_certificate(
...
+        PAGE.setStrokeColorRGB(.33,.34,.42)
+        PAGE.line(575, 175, 725, 175)
+        
+        # Add Instructor Signature
+        instructor_signature = '{0}/images/instructor.png'.format(TEMPLATE_DIR)
+        PAGE.drawImage(instructor_signature, 575, 176, 150, 37.5)
+        
+        ######################### Copied from v3 #########################
+        
+        WIDTH, HEIGHT = landscape(A4)  # Width and Height of landscape canvas (in points)
```
Comment (stvstnfrd) [inline gen_cert.py]: delete this.

File: gen_cert.py [hunk 2]
```diff
@@ -166,18 +166,19 @@ def interstitial_factory():
...
-        if template_pdf:
-            template_pdf_filename = "{0}/{1}".format(template_prefix, template_pdf)
-            if 'verified' in template_pdf:
-                self.template_type = 'verified'
-        try:
-            self.template_pdf = PdfFileReader(file(template_pdf_filename, "rb"))
-        except IOError as e:
-            log.critical("I/O error ({0}): {1} opening {2}".format(e.errno, e.strerror, template_pdf_filename))
-            raise
+        if self.template_version != "4_programmatic":
```
Comment (stvstnfrd) [inline gen_cert.py]: move this down one line and then check:

``` python
if template_pdf is not False:
   # do stuff and things
```

File: gen_cert.py [hunk 3]
```diff
@@ -1853,3 +1855,453 @@ def _generate_v3_dynamic_certificate(
...
+            ('Arial Unicode', 'Arial Unicode.ttf', None),
+        ]
+        
+        ##### Draw main rectangle containers #####
+        
+        # Outer container
+        width = 680
+        height = 488
+        PAGE.setLineWidth(2)
+        PAGE.setStrokeColorRGB(.97,.41,.12) # Orange
```
Comment (stvstnfrd) [inline gen_cert.py]: We discussed this face-to-face, but for posterity, this and related values will need to be "un-hard-coded" and extracted into `cert-data.yml`.

File: gen_cert.py [hunk 4]
```diff
@@ -1853,3 +1855,453 @@ def _generate_v3_dynamic_certificate(
...
+        certificate_string = u"CERTIFICADO DE FINALIZACIÓN"
+        (fonttag, fontfile, certificate_style) = font_for_string(
+            apply_style_to_font_list(fontlist, style_certificate_text),
+            certificate_string
+        )
+        paragraph = Paragraph(certificate_string, certificate_style)
+        certificate_width = 150
+        certificate_height = 12
+        width, height = paragraph.wrapOn(PAGE, certificate_width, certificate_height)
+        paragraph.drawOn(PAGE, 120, 472)
```
Comment (stvstnfrd) [inline gen_cert.py]: Per F2F, each related block should be extracted into a helper method like so:

``` python
paragraph = construct_alternate_text(cert_data)
paragraph.drawOn(PAGE, x, y)
```

**Comment by caesar2164** [comment]
Closing this PR, and opening another with a non-WIP branch.

**Pull request closed without merging**